masq/src/commands/financials_command.rs

```diff
@@ -58,26 +175,911 @@ impl Command for FinancialsCommand {
+                    None
+                }
+            }
+            if let Some(idx) = str.chars().position(|char| char == '.') {
+                let after_care = front_care(str, Some(idx))
+                    .chars()
+                    .rev()
+                    .skip_while(|char| *char == '0')
+                    .collect::<String>()
+                    .chars()
+                    .rev()
+                    .collect::<String>();
+                if after_care.len() != str.len() {
+                    Some(after_care)
+                } else {
+                    None
+                }
+            } else {
+                let after_care = front_care(str, None);
+                make_decision(after_care, str)
+            }
+        }
+
+        let ((time_min, time_max), (amount_min, amount_max)) = user_ranges;
+        match (
+            apply_care(time_min),
+            apply_care(time_max),
+            apply_care(amount_min),
+            apply_care(amount_max),
+        ) {
+            (None, None, None, None) => Unchanged((
+                format!("{}-{}", time_min, time_max),
+                format!("{}-{}", amount_min, amount_max),
+            )),
+            (a, b, c, d) => Changed((
+                format!(
+                    "{}-{}",
+                    a.unwrap_or_else(|| time_min.to_string()),
+                    b.unwrap_or_else(|| time_max.to_string())
+                ),
+                format!(
+                    "{}-{}",
+                    c.unwrap_or_else(|| amount_min.to_string()),
+                    d.unwrap_or_else(|| amount_max.to_string())
+                ),
+            )),
+        }
+    }
+
+    fn no_records_found<F>(stdout: &mut dyn Write, headings: &[&str], write_headings: F)
+    where
+        F: Fn(&mut dyn Write, &[&str], &[usize]),
+    {
+        let headings_widths = Self::widths_of_dynamical_headings(headings);
+        write_headings(stdout, headings, &headings_widths);
+        short_writeln!(stdout, "\nNo records found",)
+    }
+
+    fn figure_out_max_widths(records: &[&dyn WidthInfo], is_gwei: bool) -> Vec<usize> {
+        let cell_count = records[0].widths(is_gwei).len();
+        let init = vec![0_usize; cell_count];
+        let widths_except_ordinal_num = records.iter().fold(init, |acc, record| {
+            let cells = record.widths(is_gwei);
+            Self::yield_bigger_values_from_vecs(cell_count, acc, cells)
+        });
+        let mut result = vec![records.len().to_string().len()];
+        result.extend(widths_except_ordinal_num);
+        result
+    }
+
+    fn yield_bigger_values_from_vecs(
+        cell_count: usize,
+        first: Vec<usize>,
+        second: Vec<usize>,
+    ) -> Vec<usize> {
+        (0..cell_count).fold(vec![], |acc_inner, idx| {
+            plus(acc_inner, first[idx].max(second[idx]))
+        })
+    }
+
+    fn count_length_with_comma_separators<N: Display>(value: N, is_plain_integer: bool) -> usize {
```

Comment:
Just rendering this value into a temporary string, measuring the length of the string, and then throwing it away might lead to less complexity, and fewer required tests, than this function.